### PR TITLE
Build a site

### PR DESCRIPTION
**Pull request by ChristopherBiscardi**
This is all of the infrastructure and the initial implementation of a gatsby-based main.actor. You can see the site currently deployed at https://gracious-keller-8bfb46.netlify.com

image

URLs are generated as such: `packages/<name>/<version>/<markdown-file-path>` so you might end up with the following for `http`@`0.2.1` when looking at the `Payload` class:

```
packages/http/0.2.1/http/payload
```

There are some areas of different markdown files that end up looking a bit awkward (here's an example that uses types in the heading), so we'll probably want to format the output differently

image

### CODE REVIEW

**Comment by SeanTAllen** [comment]
Nice. Thanks @ChristopherBiscardi. Are you interested in working more on this going forward? Should I be trying to round up folks who might be willing to chip in?

Could you write up some "how it all works" docs that help people get started?

**Comment by SeanTAllen** [comment]
I'll merge in the next few days or so when I have some make sure the infrastructure is all in place for having it "be live".

File: build-missing-docs.sh [hunk 1]
```diff
@@ -31,10 +31,10 @@ TMPDIR=$(mktemp -d 2>/dev/null || mktemp -d -t 'tmp.XXXXXX')
 
 
 # Who we are for git
-git config --global user.email "main@ponylang.io"
-git config --global user.name "ponylang-main"
+#git config --global user.email "main@ponylang.io"
+#git config --global user.name "ponylang-main"
 # Other git setup
-git config --global push.default simple
+#git config --global push.default simple
```
Comment (SeanTAllen) [inline build-missing-docs.sh:37]: is there a reason these were commented out @ChristopherBiscardi ?

should the commenting out be reverted?

File: build-missing-docs.sh [hunk 2]
```diff
@@ -74,10 +75,11 @@ for manifest in $(find manifests -type f); do
 
       # Call the next script, responsible for building the docs.
       mkdir -p $docs_dir
-      ./build-docs.sh $code_dir $docs_dir
-      git add .
-      git commit -m "Add docs for ${name} version ${tag}"
-      git push "https://${PUSH_TOKEN}@github.com/ponylang/main.actor"
+      mkdir -p $docs_md_dir
+      ./build-docs.sh $code_dir $docs_dir $docs_md_dir
+#      git add .
```
Comment (SeanTAllen) [inline build-missing-docs.sh:80]: this should be uncommented, yes?

**Comment by SeanTAllen** [comment]
@ChristopherBiscardi do you have a netlify.toml that should be used for deploying the site on netlify?

**Comment by jemc** [comment]
Very cool!

BTW, looking through the netlify preview, it looks like there is an issue with underscores in function names being interpreted as markdown-italics:

Screenshot from 2019-07-24 07-26-26

It also looks like Sean has a typo in his function names :grin:

**Comment by jemc** [comment]
I'll also note that we might want to add a `.gitattributes` file with the following contents:
```
docs-raw-md/* linguist-generated=true
```

Doing so will cause the raw md changes to be hidden in pull requests, for easier review.

**Comment by SeanTAllen** [comment]
@jemc that .gitattributes change sounds like an excellent idea

File: build-docs.sh [hunk 3]
```diff
@@ -6,6 +6,7 @@ set -e
 # Gather expected arguments.
 CODE_DIR=$1
 DOCS_DIR=$2
+DOCS_RAW_DIR=$3
```
Comment (SeanTAllen) [inline build-docs.sh:9]: this only needs to be two args as DOCS_DIR would end up unused so the value being passed in $3 can be passed in $2 instead.

**Comment by SeanTAllen** [comment]
Closing as this has all been incorporated. Thanks Chris!

**Pull request closed without merging**